homeassistant/components/overkiz/sensor.py

```diff
@@ -101,7 +101,7 @@ class OverkizSensorDescription(SensorEntityDescription):
         name="Water consumption",
         icon="mdi:water",
         native_unit_of_measurement=UnitOfVolume.LITERS,
-        device_class=SensorDeviceClass.VOLUME,
+        device_class=SensorDeviceClass.WATER,
```

Comment:
If this one is a water meter, then the state_class should be `TOTAL` or `TOTAL_INCREASING`.

Reply:
@tetienne this would be a good one to check for other meters (like electricity) as well, not sure if everything is mapped correctly.

Reply:
@epenet Since it was introduced, I’m confused with TOTAL and TOTAL_INCREASING. I read many time the [definition](https://developers.home-assistant.io/docs/core/entity/sensor#available-state-classes), but still don’t get it.

Reply:
The name `TOTAL_INCREASING` is very confusing, we should have named it `TOTAL_AUTO_RESET` or something like that.
@tetienne did you read this part too: https://developers.home-assistant.io/docs/core/entity/sensor/#how-to-choose-state_class-and-last_reset ?

Reply:
Thx, this is indeed more clear with these examples. I asked to @gbuico to [explain](https://github.com/iMicknl/ha-tahoma/issues/708#issuecomment-1414162218) me what’s this device. But it looks like that’s not something really clear...

Reply:
Can it be that the integration is just swapping the name of the entities??
If I check the entity sensor.dhwp_actuator_water_volume_estimation_at_40_degc , this is really an "always increasing value (i.e. I have at the moment 42428 liters)
At the same time, looking at the history of sensor.dhwp_actuator_water_consumption I see values going up and done as from the chart I sent you @tetienne and this is more likely to be the estimation of remaining water at 40 degrees.

I also had a further thought on the readings of that variable and I may have found an explanation.

Problem: in my graph I see the water consumpion value float between 60 and 140 L while me heater is 65 L

My new explanation : Since my middel water temperature is around 65° it is reasonable that a 65 L heater can provide 140 L of water at 40° 

Bottom line, I think the measured values are correct... there is just a swap of names

Reply:
According to my own DHWP, `water consumption` seem to be a volume that goes down as it's "consumed" then up as it gets replenished.

![image](https://user-images.githubusercontent.com/5701372/221521674-755a7304-0dca-4de1-9965-51842b98d880.png)

In the snapshot, the steep slopes are actual hot water consumption (shower, bath, ...), the slow ones are just the water cooling down on its own inside the tank! To me this means the name of the entity is actually missleading, and should be something like `available hot water` rather than `water consumption`.

Reply:
What about the new device class VOLUME_STORAGE? It seems more accurate in this case.

Reply:
@tetienne VOLUME_STORAGE sounds fine by me! Shall we make this change and see if we can merge this one in?